skimage/segmentation/manual_segmentation.py

```diff
@@ -0,0 +1,187 @@
+        plt.draw()
+        return polygon_object
+
+    def _draw_lines(event):
+        # Do not record click events from pressing the Undo button.
+        if event.inaxes is buttonpos:
+            return
+
+        # Do not record click events when toolbar is active.
+        if fig.canvas.manager.toolbar._active is not None:
+            return
+
+        # Do not record click events outside axis.
+        if event.inaxes is None:
+            return
+
+        elif event.button == 1:  # Left click
+            temp_list.append([event.xdata, event.ydata])
+
+            # Remove previously drawn preview polygon if any.
+            if len(preview_polygon_drawn) > 0:
+                preview_polygon_drawn[-1].remove()
+                preview_polygon_drawn.remove(preview_polygon_drawn[-1])
+
+            # Preview polygon with selected vertices.
+            polygon = _draw_polygon(temp_list, alpha=alpha/1.2)
+            preview_polygon_drawn.append(polygon)
+
+        elif event.button == 2:  # Middle click
+            if len(temp_list) > 1:
+
+                # Remove the previous vertice and update preview polygon
+                temp_list.remove(temp_list[-1])
+                preview_polygon_drawn[-1].remove()
+                preview_polygon_drawn.remove(preview_polygon_drawn[-1])
+
+                polygon = _draw_polygon(temp_list, alpha=alpha/1.2)
+                preview_polygon_drawn.append(polygon)
+
+            else:
+                return
+
+        elif event.button == 3:  # Right click
+            if len(temp_list) is 0:
+                return
+
+            # Store the vertices of the polygon as shown in preview.
+            # Redraw polygon and store it in polygons_drawn so that
+            # `_undo` works correctly.
+            list_of_verts.append(list(temp_list))
+            polygon_object = _draw_polygon(list_of_verts[-1])
+            polygons_drawn.append(polygon_object)
+
+            # Empty the temporary variables.
+            preview_polygon_drawn[-1].remove()
+            preview_polygon_drawn.remove(preview_polygon_drawn[-1])
+            del temp_list[:]
+
+            plt.draw()
+
+    def _on_select(verts):
+        list_of_verts.append(verts)
+        polygon_object = _draw_polygon(verts)
+        polygons_drawn.append(polygon_object)
+        plt.draw()
+
+    def _undo(event):
+        if len(list_of_verts) > 0:
+            list_of_verts.remove(list_of_verts[-1])
+
+            # Remove previous polygon object from the plot.
+            polygons_drawn[-1].remove()
+
+            # Removes latest polygon from a list of all polygon objects.
+            # This enables undo till the first drawn polygon.
+            polygons_drawn.remove(polygons_drawn[-1])
+
+        else:
+            return
+
+    image = np.squeeze(image)
+
+    if image.ndim not in (2, 3):
+        raise TypeError('Only 2-D images or 3-D images supported.')
+
+    fig, ax = plt.subplots()
+    ax.imshow(image)
```

Comment:
Oh sorry, I misread, I thought you said if it's RGB convert to grayscale. I agree with using cmap='gray'.